### ISSUE TITLE
Question about expandable connection in MLS 3.5

### ISSUE THREAD

**Issue by Mrbuke** [modelica/ModelicaSpecification#3194]
In the section 9.1.3, MLS 3.5, we can see a description:

> - If a variable appears as an input in one expandable connector, it should appear as a non-input in at least one other expandable connector instance in the same augmentation set. An augmentation set is defined as the set of connected expandable connector instances that through the elaboration will have matching variables.

and then a example followed:

```
expandable connector EngineBus
end EngineBus; 

block Sensor
  RealOutput speed; // Output, i.e., non=input 
end Sensor ;

block Actuator
  RealInput speed; // Input 
end Actuator; 

model Engine 
  EngineBus bus; 
  Sensor sensor; 
  Actuator actuator;
equation
  connect(bus.speed, sensor.speed); // provides the noninput from sensor.speed 
  connect(bus.speed, actuator.speed); 
end Engine ;
```

I have some questions about it:

- AS the Specification said , *If a variable appears as an input in one expandable connector, it should appear as a non-input in at least one other expandable connector instance in the same augmentation set.* The `bus.speed` showed in the two connect equation in the example should be in the same expandable connector instance, right?  So what's the relation between the description and the example.

- Also as the Specification said, *[If the existing side refers to an inside connector (i.e. a connector of a component) the new variable will copy its causality, i.e. input if input and output if output, since the expandable connector must be an outside connector.]* Analyze the example only, `sensor.speed` is an inside output connector , `actuator.speed` is an inside input connector, so when they connect to `bus.speed`  respectively, which causality the `bus.speed` should select to copy?  When I am tested it in a well-known tool, the order of the two connecting equations determined the result :(

Looking forward to your answer!

**Comment by HansOlsson** [comment]
That model is too simple to illustrate what it is intended to describe. We need to have two buses, and if you try the simple example there will also be problems due to the lack of equations.

The copying of causality should be described (and implemented) in a better way. The goal is _not_ to copy the causality, but to derive a causality consistent with the restriction in 9.3.

**Comment by HansOlsson** [comment]
The text copied above is non-normative and the normative text is already correct, and has been since expandable connector was added 2.2 (it was slightly corrected in #1305). What should be clarified in the non-normative text is that it applies under certain restrictions.

As for tools:
I don't know which tool you used and found that input/output for the variable depended on the order of declaration. I do now recall that Dymola has worked like this for a long time (possibly with some weirder edge cases corrected a few years ago), and also that some user libraries depend on this.

**However**, that logic is internal - and the type-_name_ of the first connected variable is copied to the bus-variable which fooled me at first; so it might be declared as a "RealInput" but actually be an output.

**Comment by HansOlsson** [comment]
Note that even if this text wasn't updated since 2.2 it references the rules for normal connections.
That section was updated from restrictions on individual connection to restrictions on connection sets in 3.0.
